### PR TITLE
Add muting of threads.

### PR DESCRIPTION
**Pull request by charlesmchen**
Adds `mutedUntilDate` property to `TSThread`.

PTAL @michaelkirk

### CODE REVIEW

File: src/Contacts/TSThread.m [hunk 1]
```diff
@@ -10,15 +10,18 @@
...
 
 @interface TSThread ()
 
-@property (nonatomic, retain) NSDate *creationDate;
+@property (nonatomic) NSDate *creationDate;
 @property (nonatomic, copy) NSDate *archivalDate;
-@property (nonatomic, retain) NSDate *lastMessageDate;
+@property (nonatomic) NSDate *lastMessageDate;
 @property (nonatomic, copy) NSString *messageDraft;
+@property (atomic) NSDate *mutedUntilDate;
```
Comment (michaelkirk) [inline src/Contacts/TSThread.m]: nullable?

File: src/Contacts/TSThread.h [hunk 2]
```diff
@@ -145,6 +146,32 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (void)setDraft:(NSString *)draftString transaction:(YapDatabaseReadWriteTransaction *)transaction;
 
+@property (atomic, readonly) BOOL isMuted;
+@property (atomic, readonly) NSDate *mutedUntilDate;
```
Comment (michaelkirk) [inline src/Contacts/TSThread.h]: nullable?

**Review by michaelkirk — APPROVED** [review]
LGTM

File: src/Contacts/TSThread.h (hunk 2 above)
Comment (charlesmchen) [reply src/Contacts/TSThread.h]: Done.

File: src/Contacts/TSThread.m (hunk 1 above)
Comment (charlesmchen) [reply src/Contacts/TSThread.m]: Done.

**Pull request merged**